README.md

```diff
@@ -31,167 +31,118 @@ Where `<snapshot version>` refers to the version in the Sonatype Snapshot badge
 
-We can represent an Elasticsearch request as a generic data type that returns a value of type `A`. The library offers a DSL for creating these requests, by specifying their required parameters. For example, we can create a request for deleting a document with a specified index as follows:
-
-```scala
-deleteById(IndexName("index"), DocumentId("documentId"))
-```
-
-As you can see above, index names and document IDs are represented with `IndexName` and `DocumentId` respectively, using new types from ZIO Prelude, in order to increase type-safety with no runtime overhead. `IndexName` also validates the passed string according to Elasticsearch's naming criteria at compile-time using the `apply` method, or with `make` at runtime when dealing with a runtime value as an argument.
-
-All the DSL methods for request creation can be brought into scope with the following import:
+- `ElasticExecutor`: if you provide `ElasticExecutor.local`, it will run on `localhost:9200`. Otherwise, if you want to use `ElasticExecutor.live`, you must also provide `ElasticConfig`.
+- `HttpClientZioBackend`
 
 ```scala
+import sttp.client3.httpclient.zio.HttpClientZioBackend
 import zio.elasticsearch.ElasticRequest._
-```
-
-For methods receiving or returning a document of custom type `A`, you must create a schema for `A`. Here is an example of creating a schema for a custom type `EmployeeDocument`:
-
-```scala
-import zio.schema.{DeriveSchema, Schema}
-
-final case class EmployeeDocument(id: String, name: String, degree: String, age: Int)
-
-object EmployeeDocument {
-  implicit val schema: Schema[EmployeeDocument] = DeriveSchema.gen[EmployeeDocument]
+import zio.elasticsearch._
+import zio._
+
+object ZIOElasticsearchExample extends ZIOAppDefault {
+  val indexName = IndexName("index")
+  val effect: RIO[Elasticsearch, Unit] = for {
+    _ <- Elasticsearch.execute(createIndex(indexName))
+  } yield ()
+
+  override def run =
+    effect.provide(
+      ElasticExecutor.local,
+      Elasticsearch.layer,
+      HttpClientZioBackend.layer()
+    )
 }
 ```
 
-As long as we have the implicit schema value in scope, we can call the aforementioned methods, such as `getById`:
-
-```scala
-import EmployeeDocument._
-
-getById[EmployeeDocument](IndexName("index"), DocumentId("documentId"))
-```
 
-### Elastic Query
+### Type-safety with ZIO Prelude's new type
 
-In order to execute Elasticsearch query requests, both for searching and deleting by query, you first must specify the type of the query along with the corresponding parameters for that type. Queries are described with the `ElasticQuery` data type, which can be constructed from the DSL methods found under the following import:
+The library uses ZIO Prelude's new type for `IndexName`, `DocumentId` and `Routing` in order to preserve type-safety.
```

Comment:
I would write this as "ZIO Prelude's `NewType`" to make sure it's not read as a "new type for `IndexName`"

Reply:
I will replace "new type" with "Newtype". But not with `NewType`. Because `Newtype` is in ZIO Prelude. 😄